tests/ReferenceTest.php

```diff
@@ -71,6 +95,27 @@ public function testModelProperty()
         $this->assertSame('order', $o->table);
     }
 
+    public function testRefTraversing()
+    {
+        $user = new Model($this->db, 'user');
+        $user->addField('name');
+        $user->addField('company_id');
+
+        $user->id = 1;
+
+        $company = new Model($this->db, 'company');
+        $company->addField('name');
+        $user->hasOne('Company', [$company, 'our_field' => 'company_id', 'their_field' => 'id']);
+
+        $order = new Model($this->db, 'order');
+        $order->addField('company_id');
+        $order->addField('amount', ['default' => 20]);
+
+        $company->hasMany('Orders', [$order]);
+
+        $this->assertEquals(20, $user->ref('Company')->ref('Orders')->get('amount')); // 'amount' default value
```

Comment:
where is the discussed usecase?

this test seems quite complex but your usecase was about one very simple thing

this test has only one assertion for a default val

Reply:
tests expanded now from the #683

Reply:
Good. Now also need to test same things with unloaded user model.
It should return all orders of all users.

Reply:
Done